// react-native-pager-view/lib/commonjs/utils.ios.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.childrenWithOverriddenStyle = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const childrenWithOverriddenStyle = (children, pageMargin = 0) => {
  return _react.Children.map(children, child => {
    const element = child;
    return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
      style: [_reactNative.StyleSheet.absoluteFill, {
        marginRight: pageMargin / 2,
        marginLeft: pageMargin / 2
      }],
      collapsable: false,
      children: /*#__PURE__*/_react.default.cloneElement(element, {
        ...element.props,
        style: [element.props.style, _reactNative.StyleSheet.absoluteFill]
      })
    });
  });
};
exports.childrenWithOverriddenStyle = childrenWithOverriddenStyle;
